lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -777,44 +839,48 @@ struct FpToFpOpConversion
-        {{F16TyID, F8E4M3TyID}, Fp16_to_Fp8E4M3Nv},
-        {{F16TyID, F8E5M2TyID}, Fp16_to_Fp8E5M2(computeCapability >= 90)},
-        // F8 -> BF16
-        {{F8E5M2TyID, BF16TyID}, Fp8E5M2_to_Bf16(computeCapability >= 90)},
-        {{F8E4M3TyID, BF16TyID}, Fp8E4M3Nv_to_Bf16},
-        // BF16 -> F8
-        {{BF16TyID, F8E5M2TyID}, Bf16_to_Fp8E5M2(computeCapability >= 90)},
-        {{BF16TyID, F8E4M3TyID}, Bf16_to_Fp8E4M3Nv},
-        // F32 -> F8
-        {{F32TyID, F8E4M3TyID}, Fp32_to_Fp8E4M3Nv},
-        {{F32TyID, F8E5M2TyID}, Fp32_to_Fp8E5M2},
-    };
-    int inVecWidthBits = 32;
-    int outVecWidthBits = 32;
-    if (srcTy.isFloat8E4M3FNUZ() ||
-        (computeCapability >= 90 && srcTy.isFloat8E5M2() && dstTy.isF16())) {
-      inVecWidthBits = 16;
-      outVecWidthBits = 32;
-    }
-    if (dstTy.isFloat8E4M3FNUZ() ||
-        (computeCapability >= 90 && dstTy.isFloat8E5M2())) {
-      inVecWidthBits = 32;
-      outVecWidthBits = 16;
-    }
 
-    std::pair<TypeID, TypeID> key = {srcTy.getTypeID(), dstTy.getTypeID()};
+    auto undefRounding = static_cast<RoundingMode>(-1);
+
+    static DenseMap<std::tuple<TypeID, TypeID, RoundingMode>, Fp8ConvDesc>
+        srcMap = {
+            // F8 -> F16
+            {{F8E4M3B15TyID, F16TyID, undefRounding}, Fp8E4M3B15_to_Fp16},
+            {{F8E4M3FNTyID, F16TyID, undefRounding}, Fp8E4M3B15x4_to_Fp16},
+            {{F8E4M3TyID, F16TyID, undefRounding}, Fp8E4M3Nv_to_Fp16},
+            {{F8E5M2TyID, F16TyID, undefRounding},
+             Fp8E5M2_to_Fp16(computeCapability >= 90)},
+            // F16 -> F8
+            {{F16TyID, F8E4M3B15TyID, RoundingMode::RTNE},
+             Fp16_to_Fp8E4M3B15(computeCapability >= 80)},
+            {{F16TyID, F8E4M3FNTyID, RoundingMode::RTNE}, Fp16_to_Fp8E4M3B15x4},
+            {{F16TyID, F8E4M3TyID, RoundingMode::RTNE}, Fp16_to_Fp8E4M3Nv},
+            {{F16TyID, F8E5M2TyID, RoundingMode::RTNE},
+             Fp16_to_Fp8E5M2_RTNE(computeCapability >= 90)},
+            {{F16TyID, F8E5M2TyID, RoundingMode::RTZ}, Fp16_to_Fp8E5M2_RTZ},
+            // F8 -> BF16
+            {{F8E5M2TyID, BF16TyID, undefRounding},
+             Fp8E5M2_to_Bf16(computeCapability >= 90)},
+            {{F8E4M3TyID, BF16TyID, undefRounding}, Fp8E4M3Nv_to_Bf16},
+            // BF16 -> F8
+            {{BF16TyID, F8E5M2TyID, RoundingMode::RTNE},
+             Bf16_to_Fp8E5M2(computeCapability >= 90)},
+            {{BF16TyID, F8E4M3TyID, RoundingMode::RTNE}, Bf16_to_Fp8E4M3Nv},
+            // F32 -> F8
+            {{F32TyID, F8E4M3TyID, RoundingMode::RTNE}, Fp32_to_Fp8E4M3Nv},
+            {{F32TyID, F8E5M2TyID, RoundingMode::RTNE}, Fp32_to_Fp8E5M2},
+        };
+    RoundingMode rndMode =
+        roundingMode.has_value() ? roundingMode.value() : undefRounding;
+    std::tuple<TypeID, TypeID, RoundingMode> key = {srcTy.getTypeID(),
+                                                    dstTy.getTypeID(), rndMode};
```

Comment:
key = {..., roundingMode.value_or(undefRounding)}

Reply:
Thank you!!!